packages/grid/x-data-grid/src/components/containers/GridToolbarContainer.tsx

```diff
@@ -26,11 +26,12 @@ const useUtilityClasses = (ownerState: OwnerState) => {
 const GridToolbarContainerRoot = styled('div', {
   name: 'MuiDataGrid',
   slot: 'ToolbarContainer',
-  overridesResolver: (props, styles) => styles.toolbarContainer,
+  overridesResolver: (_, styles) => styles.toolbarContainer,
 })(({ theme }) => ({
   display: 'flex',
   alignItems: 'center',
   flexWrap: 'wrap',
+  gap: theme.typography.pxToRem(8),
```

Comment:
Maybe for this kind of spacing it's better to use the `spacing` api 

```suggestion
  gap: theme.spacing(2),
```